composer/spec/fixtures/packagist_responses/path_dep--path_dep.json

```diff
@@ -1 +1,198 @@
+                    </div>
+
+                <section class="wrapper">
+    <section class="container error">
+        <div class="row">
+            <h2 class="title">Oh noes, 404!</h2>
+            <h3 class="font-normal">It looks like you requested a page that was not found.</h3>
+            <h4>Go back to the <a href="/">homepage</a> or use the search above to find the package you want.</h4>
+        </div>
+    </section>
+</section>
+            </section>
+        </section>
+    </section>
+
+        <footer class="wrapper-footer">
+            <nav class="container">
+                <div class="row">
+                    <ul class="social col-xs-7 col-sm-4 col-md-2 pull-right">
+                        <li><a href="http://github.com/composer/packagist" title="GitHub"><span class="icon-github"></span></a></li>
+                        <li><a href="https://twitter.com/packagist" title="Follow @packagist"><span class="icon-twitter"></span></a></li>
+                        <li><a href="mailto:contact@packagist.org" title="Contact"><span class="icon-mail"></span></a></li>
+                    </ul>
+
+                    <ul class="col-xs-4 col-md-2">
+                        <li><a href="/about">About Packagist</a></li>
+                        <li><a href="/feeds/" rel="nofollow">Atom/RSS Feeds</a></li>
+                    </ul>
+
+                    <div class="clearfix visible-xs-block"></div>
+
+                    <ul class="col-xs-3 col-md-2">
+                        <li><a href="/statistics" rel="nofollow">Statistics</a></li>
+                        <li><a href="/explore/">Browse Packages</a></li>
+                    </ul>
+
+                    <div class="clearfix visible-xs-block"></div>
+
+                    <ul class="col-xs-3 col-md-2">
+                        <li><a href="/apidoc">API</a></li>
+                        <li><a href="/mirrors">Mirrors</a></li>
+                    </ul>
+
+                    <div class="clearfix visible-xs-block"></div>
+
+                    <ul class="col-xs-3 col-md-2">
+                        <li><a href="https://status.packagist.org/">Status</a></li>
+                    </ul>
+                </div>
+
+                <div class="row">
+                    <p class="toran col-xs-12">Packagist maintenance and hosting is provided by <a href="https://packagist.com/">Private Packagist</a></p>
+                </div>
+            </nav>
+        </footer>
+
+        <script nonce="ZMg6qEFJlrtA59ZjwkowDg==">
+            var algoliaConfig = {"app_id":"M58222SH95","search_key":"5ae4d03c98685bd7364c2e0fd819af05","index_name":"packagist"};
+                        window.process = {
+                env: { DEBUG: undefined },
+            };
+        </script>
+        <script nonce="ZMg6qEFJlrtA59ZjwkowDg==">
+    var trackPageload = false;
+</script>
+
+        <script nonce="ZMg6qEFJlrtA59ZjwkowDg==" src="/build/app.js?v=1664051059"></script>
+
+            </body>
+</html>
```

Comment:
I verified this is what `curl` sees, ie, the `html` response isn't caused by including a browser header.

I was a little surprised that the response went from the `json` `404` message to a `HTML` `404` message for a `.json` endpoint, but I suppose that makes sense if they simply changed `packagist.org` to not return anything custom so it defaults to the generic 404...

Reply:
Interestingly, I noticed:
* https://repo.packagist.org/p2/pathdep/pathdep.json -> returns `"404 not found, no packages here"`
* https://repo.packagist.org/p2/path_dep/path_dep.json | https://repo.packagist.org/p2/pathdep/path_dep.json | https://repo.packagist.org/p2/path_dep/pathdep.json all return the HTML'ified 404 page.

Not sure why the addition of the `_` results in a HTML 404 page vs a simple plain text 404 error string.

Reply:
This discrepancy is because `_` is not allowed in package names, so we route `/p2?/[A-Za-z0-9-]+/[A-Za-z0-9-]+(~dev)?\.json$` directly to filesystem to serve the static metadata fast (with a static json response for 404s) and everything else hits the application and thus returns a html 404.